src/Mutant.ts

```diff
@@ -68,93 +31,43 @@ export default class Mutant {
     this.scopedTestsById[index] = runResult;
   }
 
-  get columnNumber(): number {
-    return this.mutatedLocation.start.column + 1; //esprima starts at column 0
-  };
-
-  get filename(): string {
-    return this._filename;
-  };
-
-  get lineNumber(): number {
-    return this.mutatedLocation.start.line;
-  };
-
-  get mutatedCode(): string {
-    return this._mutatedCode;
-  };
-
-  get mutatedFilename(): string {
-    return this._mutatedFilename;
-  };
-
-  //TODO: Give this a new name. This string can be longer than a single line. Same goes for 'originalLine'.
-  get mutatedLine(): string {
-    return this._mutatedLine;
-  };
-
-  get mutator(): Mutator {
-    return this._mutator;
-  };
-
-  get originalLine(): string {
-    return this._originalLine;
-  };
-
   /**
-   * @param mutator - The Mutator which was applied to this Mutant.
+   * @param mutatorName - The name of the Mutator which created this mutant.
    * @param filename - The name of the file which was mutated, including the path.
    * @param originalCode - The original content of the file which has not been mutated.
-   * @param substitude - The mutated code which will replace a part of the originalCode.
-   * @param mutatedLocation - The part of the originalCode which has been mutated.
+   * @param replacement - The mutated code which will replace a part of the originalCode.
+   * @param location - The location of the code to be replaced
    */
-  constructor(private _mutator: Mutator, private _filename: string, private originalCode: string, substitude: string, private mutatedLocation: ESTree.SourceLocation) {
-    this.status = MutantStatus.UNTESTED;
-    this.insertSubstitude(substitude);
+  constructor(public mutatorName: string, public fileName: string, private originalCode: string, public replacement: string, public location: ESTree.SourceLocation) {
+    this.applyMutation();
   }
 
   /**
-   * Inserts the substitude into the mutatedCode based on the mutatedLocation.
-   * This also alters the originalLine and mutatedLine.
+   * Inserts the replacement into the mutatedCode based on the specified location.
    * @param substitude - The mutated code which will replace a part of the originalCode
    */
-  private insertSubstitude(substitude: string) {
+  private applyMutation() {
     let linesOfCode = this.originalCode.split('\n');
     
-    for (let lineNum = this.mutatedLocation.start.line - 1; lineNum < this.mutatedLocation.end.line; lineNum++) {
-      this._originalLine += linesOfCode[lineNum];
-      if (lineNum < this.mutatedLocation.end.line - 1) {
-        this._originalLine += '\n';
+    this.originalLines = '';
+    for (let lineNum = this.location.start.line - 1; lineNum < this.location.end.line; lineNum++) {
+      this.originalLines += linesOfCode[lineNum];
+      if (lineNum < this.location.end.line - 1) {
+        this.originalLines += '\n';
```

Comment:
Well.. we could.. but i don't feel like splitting and concatenating all the time.

We also could add the *index based range* to the esprima parsing. See http://esprima.org/demo/parse.html and check *index based range*. It will add a `range` next to the current `loc`:

```json
"range": [
        35,
        54
    ],
```

This will be much easier for finding and replacing code. I think the performance impact will be low. Maybe we could add a story for that? That way we can remove most of the `split('\n\)` from our code.